src/overlay/OverlayManagerImpl.h

```diff
@@ -79,7 +83,7 @@ class OverlayManagerImpl : public OverlayManager
     PeerManager mPeerManager;
     PeerDoor mDoor;
     PeerAuth mAuth;
-    bool mShuttingDown;
+    std::atomic<bool> mShuttingDown;
```

Comment:
I don't think this has to be atomic anymore, does it? Nobody else besides main thread should observe it even through methods.

Reply:
I think `shouldAbort` should still check if overlay is shutting down, similar to what we do in master